sljit_src/sljitNativePPC_common.c

```diff
@@ -1424,12 +1437,24 @@ static sljit_s32 emit_rev(struct sljit_compiler *compiler, sljit_s32 op,
 			offs_reg = TMP_REG2;
 		}
 #if (defined SLJIT_CONFIG_PPC_64 && SLJIT_CONFIG_PPC_64)
-	} else if (memw > 0x7fff7fffl || memw < -0x80000000l) {
+#if !(defined(_ARCH_PWR7) && _ARCH_PWR7)
+		if ((op == SLJIT_REV) && !is_32 && (TMP_MEM_START != 0))
+			FAIL_IF(push_inst(compiler, ADDI | D(offs_reg) | A(offs_reg) | IMM(TMP_MEM_START)));
```

Comment:
The question I raised still stands.

big endian:
`swap[base,offset] = reg; offset += 4; tmp =reg >> 32; swap[base,offset] = tmp`
little endian:
`tmp =reg >> 32; swap[base,offset] = tmp; offset += 4; swap[base,offset] = reg`

Therefore we don't need to add an extra instruction for little endian, just reorder the operations. Is there a reason why this cannot be done?

Reply:
Just wish it would look as clean as the MIPS code, but at least most of it won't be executed by default since POWER7 is more than 10 years old and targeted by compilers by default

Reply:
I would not mind moving to a newer minimum requirement. I need to figure out how to use this with qemu. I don't even understand the difference between powerpc and power.

Reply:
the additional instructions are enabled by the compiler, so first need to compile (assuming `gcc` and POWER10 for example) with `-mcpu=power10`

`qemu` has a `-cpu` parameter that indicates the CPU type it is to emulate so there it would be `-cpu power10` assuming you have a recent enough version.

you can get the full list with `-cpu help`